packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -344,9 +360,10 @@ export function createCommentsRoom<TThreadMetadata extends BaseMetadata>(
 
     mutate(room.editThreadMetadata({ metadata, threadId }), {
       optimisticData,
-    }).catch((err: Error) => {
+    }).catch((err: CommentsApiError) => {
+      const error = handleCommentsApiError(err);
```

Comment:
TypeScript will force you to use `err: unknown` in normal catch statements, and for good reason. Since anything can be thrown at any time, there is no way to guarantee that at runtime this value will actually be a `CommentsApiError` instance.

Therefore, the safer way to write this is:
```suggestion
    }).catch((err: unknown) => {
      if (!(err instanceof CommentsApiError)) { throw e; }
      const error = handleCommentsApiError(err);
```

…or some variation thereof.

Reply:
As a general principle (maybe even a lint rule?), the `err` argument of a `.catch()` handler must always we typed as `unknown`.